src/components/Envoy/EnvoyModal.tsx

```diff
@@ -70,17 +70,14 @@ class EnvoyDetailsModal extends React.Component<EnvoyDetailProps, EnvoyDetailSta
     this.state = {
       config: {},
       resource: 'all',
-      fetch: true,
+      fetch: false,
       pod: this.sortedPods()[0]
     };
   }
 
-  componentDidMount() {
-    this.fetchContent();
-  }
-
-  componentDidUpdate() {
-    if (this.state.fetch) {
+  componentDidUpdate(prevProps: EnvoyDetailProps) {
+    // Fetch data when the modal has just opened or some action request a fetch.
+    if ( (!prevProps.show && this.props.show) || this.state.fetch) {
```

Comment:
just a suggestion, that perhaps would make the things simpler: if you remove this `show` props, and in the parent component `WorkloadWizardDropdown` you only render this component when it has to be shown, then i guess it makes everything easier and you can go back having the backend call in `componentDidMount`, no?